lua/telescope/builtin/init.lua

```diff
@@ -322,4 +352,50 @@ builtin.lsp_document_diagnostics = require('telescope.builtin.lsp').diagnostics
 ---@field hide_filename boolean: if true, hides the name of the file in the current picker (default is false)
 builtin.lsp_workspace_diagnostics = require('telescope.builtin.lsp').workspace_diagnostics
 
+local apply_config = function(mod)
+  local pickers_conf = require('telescope.config').pickers
+  for k, v in pairs(mod) do
+    local pconf = vim.deepcopy(pickers_conf[k] or {})
+    if pconf.theme then
+      local theme = pconf.theme
+      pconf.theme = nil
+      pconf = require("telescope.themes")["get_" .. theme](pconf)
+    end
+    if pconf.mappings then
+      local mappings = pconf.mappings
+      pconf.mappings = nil
+      pconf.attach_mappings = function(_, map)
+        for mode, tbl in pairs(mappings) do
+          for key, action in pairs(tbl) do
+            if type(action) == "string" then
+              map(mode, key, require("telescope.actions")[action])
```

Comment:
we should assert that this exists: `map(mode, key, assert("telescope.actions")[action], string.format("'%s' is not a valid action", action))`

Reply:
We kinda do in the `telescope.actions` table. You will get `error("Key does not exist for 'telescope.actions': " .. tostring(k))`.

If you want something additional here we can also do that

Reply:
Ah, nice, I forgot that we already did that. Then nevermind :)